docs/design-time-builds.md

```diff
@@ -1,186 +1,214 @@
 # Design-time builds
 
-- [What is a design-time build?](#what-is-a-design-time-build)
-- [Targets that run during design-time builds](#targets-that-run-during-design-time-builds)
-- [Designing targets for use in design-time builds](#designing-targets-for-use-in-design-time-builds)
-- [Diagnosing design-time builds](#diagnosing-design-time-builds)
+## What is a design-time build
 
-## What is a design-time build?
+Visual Studio needs various information about your projects, including source files, references and options. In .NET, much of this data is defined in MSBuild `<Project>` files, such as `MyProject.csproj`. Visual Studio uses the MSBuild engine to obtain data about a project via a so-called _design-time build_ (DTB).
 
-Design-time builds are special builds that are launched by the project system to gather just enough information to populate the language service and other project services, such as the Dependencies node.  Design-time builds are not directly user-initiated, but may be indirectly launched in response to a user action such as changing the project file, build options, adding/removing source files and references, or switching configurations.
+DTBs differ from normal builds in a few key ways:
 
-For performance reasons, and unlike normal builds which call the _Build_ target, design-time builds call a limited set of targets. This can lead to custom builds that succeed during a normal build, but end up failing during a design-time build, typically due to custom targets with under-specified dependencies.
+- They are scheduled automatically by VS, such as during project load or when a project file changes.
+- They run behind-the-scenes, without showing output in the IDE (generally).
+- No compilation occurs, and outputs aren't modified.
+- Additional MSBuild targets are invoked to obtain various data.
 
-## Targets that run during design-time builds
+## What features do design-time builds support
```

Comment:
nit: change to "Features that design-time builds support" or add question mark to end, I was confused what this meant for a sec

Reply:
I left off question marks intentionally as I'm pretty sure that's the norm in MS tech docs. Chatty seems to agree: https://chatgpt.com/share/673d270b-9474-8004-98c7-3d9181863c5e

But I couldn't find a reference for this in the [MS Style Guide](https://learn.microsoft.com/en-us/style-guide/).

Reply:
Oh, I found one. From https://learn.microsoft.com/en-us/style-guide/scannable-content/headings#formatting-headings:

> Don't end headings with a period. A question mark or (rarely) an exclamation point can be used if it's needed for meaning.
> Examples
> Not seeing what you want?
> What can we help you find?

I'll revisit this. Thanks!

Reply:
Fixed.